src/commands/index.rs

```diff
@@ -209,27 +216,34 @@ where
         .ok_or("ERR no such index".into())
         .map(|schema| &schema.index)
         .and_then(|index| {
-            let result: Value =
+            let result =
                 index
                     .search(&query)?
                     .try_fold(Value::Object(Map::new()), |mut acc, key| {
-                        ctx.open_key(&key)
-                            .get_value::<RedisJSON>(&REDIS_JSON_TYPE)
-                            .and_then(|doc| {
-                                doc.map_or(Ok(Vec::new()), |data| {
-                                    data.get_values(&path)
-                                        .map_err(|e| e.into()) // Convert Error to RedisError
-                                        .map(|values| {
-                                            values.into_iter().map(|val| val.clone()).collect()
-                                        })
+                        let redis_key = ctx.open_key(&key);
+                        if redis_key.is_null() {
+                            // remove doc that doesn't exist from index
+                            remove_document(&key, &index_name)?;
+                            Ok(acc)
+                        } else {
```

Comment:
Please explain when this can happen, if we remove it from the index when deleting. Is this for backwards compatibility?

Reply:
it was the original lazy solution, I think it can be removed now.